metadata-ingestion/src/datahub/ingestion/source/redshift/lineage.py

```diff
@@ -130,6 +167,73 @@ def __init__(
             self.report.lineage_end_time,
         ) = self.get_time_window()
 
+    def _init_temp_table_schema(
+        self, database: str, temp_tables: List[TempTableRow]
+    ) -> None:
+        if self.context.graph is None:  # to silent lint
+            return
+
+        schema_resolver: sqlglot_l.SchemaResolver = (
+            self.context.graph._make_schema_resolver(
+                platform=LineageDatasetPlatform.REDSHIFT.value,
+                platform_instance=self.config.platform_instance,
+                env=self.config.env,
+            )
+        )
+
+        dataset_vs_columns: Dict[str, List[SchemaField]] = {}
+        # prepare dataset_urn vs List of schema fields
+        for table in temp_tables:
+            result = sqlglot_l.create_lineage_sql_parsed_result(
```

Comment:
Why don't you cache the temp tables here? 
In your current code you run query parser twice on each temp table ddl. 
Here, you get the query type as well, so you can cache the real temp table ddls

Reply:
will make the temp_table as instance variable